### ISSUE TITLE
io.netcdf masking should use masked_equal rather than masked_value

### ISSUE THREAD

**Issue by billsacks** [scipy/scipy#5626]
Handling of masked values (\_FillValue / missing\_value) was added to io.netcdf in PR #5228 . However, I'm wondering if np.ma.masked_values is really the right function to use in \_\_getitem\_\_. I think masked_equal would be more appropriate here, since masked_values masks any value that is "close" to the given \_FillValue. Most (all?) netcdf utilities I have worked with only apply the \_FillValue to array values that are identical to the given \_FillValue (down to the last bit).

For example:

```
x = [0, 1e-10, 1e-5]

xm = ma.masked_values(x, 0)

xm
Out[390]: 
masked_array(data = [-- -- 1e-05],
             mask = [ True  True False],
       fill_value = 0.0)
```

I would expect the second array element to NOT be masked here.

**Comment by efiring** [comment]
Good point.  The Unidata netCDF4 interface uses a test for equality, and also takes care to handle NaN as a missing value.  Would you like to submit a PR?

**Comment by billsacks** [comment]
Yes, I'm happy to give this a shot. It might be a little while, given the holidays and the fact that I've never contributed to scipy before (so I'll need to familiarize myself with the process). But I'll try to get this done in the next couple of weeks.

I'm not familiar with the netCDF4 or scipy licenses: Do you know if there is anything I should be aware of before borrowing code from netCDF4?

**Comment by efiring** [comment]
https://github.com/Unidata/netcdf4-python/blob/master/COPYING
I think we can work with this by including the notice and copyright somewhere, with a note as to what it applies to, but someone like @rgommers would know more about this.

**Comment by rgommers** [comment]
That looks BSD compatible, so the main decision is where to put the license info. I'd suggest to put it in the file itself, similar to this: https://github.com/scipy/scipy/blob/master/scipy/special/cephes/scipy_iv.c#L32

The requirement to put that same license notice in "supporting documentation" is phrased a bit oddly and can be interpreted in several ways. I think the amount of code we want to use from netcdf4-python is quite minor, and putting a new section in the scipy html reference guide for that doesn't really make sense to me. So I'd suggest just the copyright statement in the source code itself. Pinging @jswhit to see if he sees a problem with that.

**Comment by billsacks** [comment]
@efiring : I have addressed this issue in #5640 . I have also made a few other changes to the handling of masks, which I describe in that PR. I would welcome any feedback you have on this.

@rgommers : Thank you for your thoughts on copying code from netcdf4. As I mention in #5640 I ended up taking very little code from netcdf4 (more inspiration than code), so it didn't feel necessary to me to include this information for now, but I can do so if others think I should. In any case, this is helpful information to have for the future.
